app/controllers/successes_controller.rb

```diff
@@ -1,6 +1,6 @@
 class SuccessesController < ApplicationController
-  def index 
-    coordinates_retriever = GoogleMapCoordinates.new
-    @map_collection = coordinates_retriever.generate_coordinates
+  def index
```

Comment:
I think this class is doing more than just coordinates it's it? It's preparing coordinates from `Location` and data from `Dog` in order to prepare metadata for google maps. Maybe we could name it `google_maps_metadata`?

Reply:
I agree. This problem will be resolved once the logic is separated to service object.